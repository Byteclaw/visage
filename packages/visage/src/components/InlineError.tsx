import { markAsVisageComponent, VisageComponent } from '@byteclaw/visage-core';
import React, { ReactNode } from 'react';
import { Flex } from './Flex';
import { ErrorIcon } from '../assets';
import { SvgIcon } from './SvgIcon';
import { StyleProps } from '../types';

interface InlineErrorProps {
  children: ReactNode;
}

export const InlineError: VisageComponent<InlineErrorProps, StyleProps> = ({
  children,
  ...restProps
}: InlineErrorProps) => {
  return (
    <Flex
      {...restProps}
      styles={{ color: 'danger', fontSize: 0, lineHeight: 0 }}
    >
      <SvgIcon icon={ErrorIcon} styles={{ mr: 1 }} />
      {children}
    </Flex>
  );
};

InlineError.displayName = 'InlineError';

markAsVisageComponent(InlineError);
