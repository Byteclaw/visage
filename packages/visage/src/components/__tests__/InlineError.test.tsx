import React from 'react';
import { createTestTheme, render } from './render';
import { InlineError } from '../InlineError';

describe('InlineError', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<InlineError>This is an error</InlineError>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by InlineError face', () => {
    const { asFragment } = render(<InlineError>This is an error</InlineError>, {
      ds: {
        theme: createTestTheme({
          InlineError: {
            color: 'blue',
          },
        }),
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by InlineErrorIcon face', () => {
    const { asFragment } = render(<InlineError>This is an error</InlineError>, {
      ds: {
        theme: createTestTheme({
          InlineErrorIcon: {
            color: 'blue',
          },
        }),
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by styles prop', () => {
    const { asFragment } = render(
      <InlineError styles={{ color: 'blue' }}>This is an error</InlineError>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
