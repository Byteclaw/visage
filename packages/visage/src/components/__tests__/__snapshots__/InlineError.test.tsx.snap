// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineError is extendable by InlineError face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  margin-right: 8px;
}

.emotion-0::before {
  content: "\\200b";
}

.emotion-0 svg {
  height: 1em;
  vertical-align: middle;
  width: auto;
}

.emotion-1 {
  color: rgb(0,0,255);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  line-height: inherit;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      <svg
        focusable="false"
        viewBox="0 0 20 20"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M-2-2h24v24H-2z"
          />
          <path
            d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    This is an error
  </div>
</DocumentFragment>
`;

exports[`InlineError is extendable by InlineErrorIcon face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-1 {
  color: rgb(255,0,0);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  line-height: inherit;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  margin-right: 8px;
  color: rgb(0,0,255);
}

.emotion-0::before {
  content: "\\200b";
}

.emotion-0 svg {
  height: 1em;
  vertical-align: middle;
  width: auto;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      <svg
        focusable="false"
        viewBox="0 0 20 20"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M-2-2h24v24H-2z"
          />
          <path
            d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    This is an error
  </div>
</DocumentFragment>
`;

exports[`InlineError is extendable by styles prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  margin-right: 8px;
}

.emotion-0::before {
  content: "\\200b";
}

.emotion-0 svg {
  height: 1em;
  vertical-align: middle;
  width: auto;
}

.emotion-1 {
  color: rgb(0,0,255);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  line-height: inherit;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      <svg
        focusable="false"
        viewBox="0 0 20 20"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M-2-2h24v24H-2z"
          />
          <path
            d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    This is an error
  </div>
</DocumentFragment>
`;

exports[`InlineError renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-1 {
  color: rgb(255,0,0);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  line-height: inherit;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  margin-right: 8px;
}

.emotion-0::before {
  content: "\\200b";
}

.emotion-0 svg {
  height: 1em;
  vertical-align: middle;
  width: auto;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      <svg
        focusable="false"
        viewBox="0 0 20 20"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M-2-2h24v24H-2z"
          />
          <path
            d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    This is an error
  </div>
</DocumentFragment>
`;
